osu.Game/Seasonal/MainMenuSeasonalLighting.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Timing;
+using osu.Framework.Utils;
+using osu.Game.Audio;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Seasonal
+{
+    public partial class MainMenuSeasonalLighting : CompositeDrawable
+    {
+        private IBindable<WorkingBeatmap> working = null!;
+
+        private InterpolatingFramedClock beatmapClock = null!;
+
+        private List<HitObject> hitObjects = null!;
+
+        private RulesetInfo? osuRuleset;
+
+        private int? lastObjectIndex;
+
+        public MainMenuSeasonalLighting()
+        {
+            // match beatmap playfield
+            RelativeChildSize = new Vector2(512, 384);
+
+            RelativeSizeAxes = Axes.X;
+
+            Anchor = Anchor.Centre;
+            Origin = Anchor.Centre;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(IBindable<WorkingBeatmap> working, RulesetStore rulesets)
+        {
+            this.working = working.GetBoundCopy();
+            this.working.BindValueChanged(_ => Scheduler.AddOnce(updateBeatmap), true);
+
+            // operate in osu! ruleset to keep things simple for now.
+            osuRuleset = rulesets.GetRuleset(0);
```

Comment:
This causes an issue when testing custom rulesets using the OsuGame test scene, where the osu ruleset doesn't exist by default, causing a null reference later on.

Reply:
Pray tell where the nullref is?

Reply:
Added some extra safeties, please re-test.

Reply:
Oops. Turns out I somehow was building against the nuget package, instead of this branch, which did have the issue. Upon double checking it seems that this was already fixed in `updateBeatmap()` prior to the extra safeties added. https://github.com/ppy/osu/blob/d897a31f0c5b63534f60d165857bd67123a854e4/osu.Game/Seasonal/MainMenuSeasonalLighting.cs#L63-L68

Checking out the branch as is, I no longer experience a crash.
Sorry for the false alarm, my bad on that one.